packages/server/src/course/course.controller.ts

```diff
@@ -893,4 +896,84 @@ export class CourseController {
       });
     return;
   }
+
+  // UPDATE course_settings_model SET selectedFeature = false WHERE courseId = selectedCourseId;
+  // will also create a new course settings record if it doesn't exist for the course
+  @Patch(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async enableDisableFeature(
+    @Param('id') courseId: number,
+    @Body() body: CourseSettingsRequestBody,
+  ): Promise<void> {
+    // fetch existing course settings
+    let courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId: courseId },
+    });
+
+    // if no course settings exist yet, create new course settings for the course
+    if (!courseSettings) {
+      // first make sure the course exists in course table (this check might not be needed as the guards already make sure the user is in the course (therefore the course must exist), but this is a rare function to be called so the small performance hit is acceptable for later safety)
+      const course = await CourseModel.findOne({
+        where: { id: courseId },
+      });
+      if (!course) {
+        throw new NotFoundException(
+          'Error while creating course settings: Course not found',
+        );
+      }
+      courseSettings = new CourseSettingsModel(); // all features are enabled by default, adjust in CourseSettingsModel as needed
+      courseSettings.courseId = courseId;
+    }
+
+    // then, toggle the requested feature
+    try {
+      courseSettings[body.feature] = body.value;
+    } catch (err) {
+      throw new BadRequestException('Invalid feature');
+    }
+
+    try {
+      await courseSettings.save();
+    } catch (err) {
+      throw new HttpException(
+        'Error while saving course settings',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  // SELECT * FROM course_settings_model WHERE courseId = selectedCourseId;
+  // if no settings for the courseid, return all true
+  @Get(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR, Role.STUDENT, Role.TA)
+  async getFeatures(
+    @Param('id') courseId: number,
+  ): Promise<CourseSettingsResponse> {
+    const courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId },
+    });
+
+    // if no settings found, do default
+    if (!courseSettings) {
```

Comment:
If there is no settings for the course, we should throw the error? The expectation is that when we create a course the course settings table should be populated?

Reply:
I thought about this, and I realised that yes while the expectation is there should be a corresponding courseSettings model, I also realised that it's probably not worth it to potentially crash the application just because a relatively minor feature isn't working correctly. 

I originally wanted the method to give a 202 Accepted if no courseSettings was found, but out of fear that using res could break the type safety somehow (this was my first backend method), I instead opted for returning a "settingsFound" variable.